pontoon/sync/vcs/repositories.py

```diff
@@ -71,15 +71,41 @@ def pull(self, source=None, target=None, branch=None):
 class PullFromHg(PullFromRepository):
 
     def pull(self, source=None, target=None):
-        log.debug("Mercurial: Update repository.")
-
         source = source or self.source
         target = target or self.target
 
-        # Undo local changes: Mercurial doesn't offer anything more elegant
-        command = ["rm", "-rf", target]
-        code, output, error = execute(command)
+        if os.path.exists(target):
+            log.debug("Mercurial: Update repository.")
+
+            # Make sure hg extensions needed to undo local changes are enabled
+            with open(os.path.join(target, '.hg/hgrc'), 'a+') as hgrc:
+                pontoon_hook = '# pontoon-extensions\n'
+                hgrc.seek(0)
+                if not pontoon_hook in hgrc.read():
+                    hgrc.write('\n[extensions]\n')
+                    hgrc.write('strip =\n')
+                    hgrc.write('purge =\n')
+                    hgrc.write(pontoon_hook)
+
+            # Undo local changes: discard uncommitted changes
+            command = ["hg", "update", "--clean"]
+            execute(command, target)
+
+            # Undo local changes: delete untracked files
+            command = ["hg", "purge"]
+            execute(command, target)
+
+            # Undo local changes: discard local commits
+            command = ["hg", "strip", "'roots(outgoing())'"]
+            execute(command, target)
+
+            command = ["hg", "pull", "-u"]
+            code, output, error = execute(command, target)
+
+            if code == 0:
```

Comment:
Should we raise an error if code != 0? I think it would be good to log what happened